tenant-isolation-config-service-impl/src/main/java/org/hypertrace/tenantisolation/config/service/TenantIsolationConfigServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.hypertrace.tenantisolation.config.service;
+
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationConfigStore;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationConfigStoreProvider;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationGroupConfigDTO;
+import org.hypertrace.tenantisolation.config.service.v1.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TenantIsolationConfigServiceImpl
+    extends TenantIsolationConfigServiceGrpc.TenantIsolationConfigServiceImplBase {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(TenantIsolationConfigServiceImpl.class);
+
+  private static final String GENERIC_CONFIG_SERVICE = "generic.config.service";
+
+  private TenantIsolationConfigServiceRequestValidator validator =
+      new TenantIsolationConfigServiceRequestValidator();
+
+  private TenantIsolationConfigStore tenantIsolationConfigStore;
+
+  @Inject
+  public TenantIsolationConfigServiceImpl(Config config) {
+    Config cfg = config.getConfig(GENERIC_CONFIG_SERVICE);
+    this.tenantIsolationConfigStore = TenantIsolationConfigStoreProvider.getDocumentStore(cfg);
```

Comment:
Prefer injecting this especially - makes testing much nicer to be able to mock it.